airbyte-ci/connectors/pipelines/pipelines/gradle.py

```diff
@@ -37,7 +38,10 @@ def __init__(self, context: PipelineContext) -> None:
 
     @property
     def connector_java_build_cache(self) -> CacheVolume:
-        return self.context.dagger_client.cache_volume("gradle-cache")
+        # TODO: remove this once we finish the project to boost source-postgres CI performance.
+        # We should use a static gradle-cache volume name.
+        cache_volume_name = hacks.get_gradle_cache_volume_name(self.context, self.logger)
```

Comment:
Assuming that these volumes outlive the pipeline execution (which I have doubts about) isn't this dangerous? Won't this leave a bunch of garbage lying around in the dagger engine?

Irrespective of whether I'm right or not about cache volume persistence, it's simpler to reimplement this PR by modifying `_get_gradle_command` instead:

```
    def _get_gradle_command(self, task: str) -> List[str]:
        return sh_dash_c(
            [
                # The gradle command is chained in between a couple of rsyncs which load from- and store to the cache volume.
                # TODO: remove test after Q4a2023 (github issue #123456)
                f"(test '{self.context.connector.technical_name}' != 'source-postgres' && rsync -a --stats /root/gradle-cache/ /root/.gradle || true)",
                f"./gradlew {' '.join(self.DEFAULT_GRADLE_TASK_OPTIONS)} {task}",
                f"(test '{self.context.connector.technical_name}' != 'source-postgres' && rsync -a --stats /root/.gradle/ /root/gradle-cache || true)",
            ]
        )
```

I can't foresee us wanting to disable caching for anything other than source-postgres.

Reply:
What we want to avoid to get reproducible metrics is the persistent caching which could potentially make other runs feed the cache and bias the results on master.
 I we use a completly different cache volume but keep rsync we get unbiased performance but keep the intra-pipeline gradle caching which is working correctly today. 

The volumes will outlive pipeline execution, but the dagger-engine on our infra is run on a K8S that is not supposed to idle forever. In other word we have disk renewal.

Reply:
Ah, you're right. OK!